reflection-doc/src/service.rs

```diff
@@ -15,6 +16,12 @@ use crate::{
 };
 use reflection_node::Node;
 
+#[derive(Error, Debug)]
+pub enum StartupError {
+    #[error(transparent)]
+    Node(#[from] anyhow::Error),
+}
```

Comment:
This wraps an generic error type in an concrete error type. I suggest to either convert from an actual (external) error type (for example coming from `reflection-node`) and if it doesn't exist (or is already generic) skip `thiserror` and just use `anyhow` to handle it.

`thiserror` should be used for defining concrete error types (for example for clear APIs) and `anyhow` for consuming them (for example an application using that API).

Reply:
I do prefer concrete error types, even though I can be convinced of using anyhow, if there are good reasons. Currently, the node uses anyhow, and I didn't want to change that inside this PR. If you want, I can add a commit removing anyhow form node.

Reply:
> I do prefer concrete error types, even though I can be convinced of using anyhow, if there are good reasons. Currently, the node uses anyhow, and I didn't want to change that inside this PR. If you want, I can add a commit removing anyhow form node.

For libaries I also prefer not using `anyhow`, for applications I don't see much reason unless the application actually really needs to handle errors differently depending on the type, otherwise one ends up with a lot of type overhead for no used functionality. But that's up to you to decide, I'm fine with either.